arrow/src/compute/kernels/sort.rs

```diff
@@ -1442,7 +1444,7 @@ mod tests {
                 nulls_first: true,
             }),
             Some(3),
-            vec![None, None, Some(2)],
+            vec![None, Some(2), Some(0)],
```

Comment:
I think the checked in answers are incorrect -- namely the first three items in the input are `None`, `0` and `2` and yet the output has two nulls in it.

Reply:
Is this passing? This looks wrong to me, the unoptimized order (ie /wo limit) would look like `None, None, 2, 0, 0, -1`, applying the limit: `None, None, 2`.